File: src/java/org/waxeye/parser/State.java
/*
 * Waxeye Parser Generator
 * www.waxeye.org
 * Copyright (C) 2008-2010 Orlando Hill
 * Licensed under the MIT license. See 'LICENSE' for details.
 */
package org.waxeye.parser;

import java.util.List;

/**
 * An automaton state.
 *
 * @param <E> The node types for the AST.
 *
 * @author Orlando Hill
 */
public final class State <E extends Enum<?>>
{
    /** The edges of the state. */
    private final List<Edge<E>> edges;

    /** Whether the state is a matching state. */
    private final boolean match;

    /**
     * Creates a new State.
     *
     * @param edges The edges of the state.
     *
     * @param match Whether the state is a matching state.
     */
    public State(final List<Edge<E>> edges, final boolean match)
    {
        this.edges = edges;
        this.match = match;
    }

    /**
     * Returns the edges.
     *
     * @return Returns the edges.
     */
    public List<Edge<E>> getEdges()
    {
        return edges;
    }

    /**
     * Returns the match.
     *
     * @return Returns the match.
     */
    public boolean isMatch()
    {
        return match;
    }
}

File: src/php/parser/State.php
<?php


namespace parser;


use JsonSerializable;

class State implements JsonSerializable
{
    private Edges $edges;
    private bool $match;

    public function __construct(Edges $edges, bool $match)
    {
        $this->edges = $edges;
        $this->match = $match;
    }

    /**
     * @return Edges
     */
    public function getEdges(): Edges
    {
        return $this->edges;
    }

    /**
     * @return bool
     */
    public function isMatch(): bool
    {
        return $this->match;
    }

    public function __toString()
    {
        return json_encode($this);
    }

    /**
     * @inheritDoc
     */
    public function jsonSerialize()
    {
        return array("State" => get_object_vars($this));
    }
}


